### DISCUSSION TITLE
(❓) Use GitHub releases as a source

### DISCUSSION THREAD

**Discussion by KotlinIsland** [python-poetry/poetry#9517] [Q&A] [answered]
# pip be like
```
pip install --upgrade mypy --pre --find-links https://github.com/mypyc/mypy_mypyc-wheels/releases
...
Successfully installed mypy-1.11.0+dev
```

# poetry be like
```
> poetry source add mypy https://github.com/mypyc/mypy_mypyc-wheels/releases
> poetry add mypy

Could not find a matching version of package mypy
```

How can I ditch `pip` for this use case and specify arbitrary GitHub releases as sources?

**Comment by Secrus** [comment]
Poetry doesn't support `--find-links` from `pip`. Sources used with Poetry must follow the PEP 503 Simple API protocol.

**Reply by KotlinIsland** [reply]
So it's currently impossible to use GitHub releases? Could a plugin resolve this?

**Reply by Secrus** [reply, marked as answer]
You can only use GH releases if you point to the exact file you want to use.

A plugin could solve this, but you would have to dig into the internals of the repositories system to figure out how to do it.
